lib/resty/dns/resolver.lua

```diff
@@ -557,6 +558,27 @@ local function parse_section(answers, section, buf, start_pos, size,
 
             ans.ptrdname = name
 
+        elseif typ == TYPE_SOA then
+            local name, p = _decode_name(buf, pos)
+            if not name then
+                return nil, pos
+            end
+            ans.mname = name
+
+            pos = p
+            name, p = _decode_name(buf, pos)
+            if not name then
+                return nil, pos
+            end
+            ans.rname = name
+
+            for _, name in ipairs(soa_names) do
```

Comment:
Here we reference the global variable `ipairs`, which can be caught by the `lj-releng` tool (with some minor modifications). We should add the line `local ipairs = ipairs` to the outer scope to avoid this.